packages/next/src/server/web/sandbox/context.ts

```diff
@@ -238,30 +240,35 @@ Learn More: https://nextjs.org/docs/messages/edge-dynamic-code-evaluation`),
           init.headers.set(`user-agent`, `Next.js Middleware`)
         }
 
-        if (typeof input === 'object' && 'url' in input) {
-          return __fetch(input.url, {
-            ...pick(input, [
-              'method',
-              'body',
-              'cache',
-              'credentials',
-              'integrity',
-              'keepalive',
-              'mode',
-              'redirect',
-              'referrer',
-              'referrerPolicy',
-              'signal',
-            ]),
-            ...init,
-            headers: {
-              ...Object.fromEntries(input.headers),
-              ...Object.fromEntries(init.headers),
-            },
-          })
-        }
-
-        return __fetch(String(input), init)
+        const response =
+          typeof input === 'object' && 'url' in input
+            ? __fetch(input.url, {
+                ...pick(input, [
+                  'method',
+                  'body',
+                  'cache',
+                  'credentials',
+                  'integrity',
+                  'keepalive',
+                  'mode',
+                  'redirect',
+                  'referrer',
+                  'referrerPolicy',
+                  'signal',
+                ]),
+                ...init,
+                headers: {
+                  ...Object.fromEntries(input.headers),
+                  ...Object.fromEntries(init.headers),
+                },
+              })
+            : __fetch(String(input), init)
+
+        return await response.catch((err) => {
+          callingError.message = err.message
+          err.stack = callingError.stack
+          throw err
```

Comment:
@jridgewell I tried going with this path instead of string manipulation (which felt hacky to me as it requires a message to be in a single line etc).

wdyt about this?
I also added an example to test.

If that's good enough yet we can have improvements, let's consider them in a different PR?
I want to land this because right now `fetch` stack traces are bad, and improvements can always have follow-ups <picture data-single-emoji=":blob_smile:" title=":blob_smile:"><img class="emoji" width="20" height="auto" src="https://single-emoji.vercel.app/api/emoji/eyJhbGciOiJkaXIiLCJlbmMiOiJBMjU2R0NNIn0..WFd4Emd22wkAbsiK.Pwo45P_OKvj_OVTvFHzfd1dVQeKwUXkY-YKAMdTG4YXCNWjMw0aaVTyaPj3RumQiMAh9iXv-CffeqrBu1j89KcXq_T1usS3gMU-lwCN-alz64qT1hSi6W6QGPZs.TX4VM2aFOzwBSJcN_00Uxw" alt=":blob_smile:" align="absmiddle"></picture>